.github/workflows/pwa-deployment.yml

```diff
@@ -0,0 +1,492 @@
+            fi
+          else
+            # Production/staging deployment to root
+            echo "s3-prefix=" >> $GITHUB_OUTPUT
+            
+            if [ -n "${{ inputs.preview-base-url }}" ]; then
+              echo "deployment-url=${{ inputs.preview-base-url }}/" >> $GITHUB_OUTPUT
+            else
+              echo "deployment-url=https://${{ inputs.s3-bucket }}.s3.amazonaws.com/index.html" >> $GITHUB_OUTPUT
+            fi
+          fi
+
+          if [ "${{ inputs.debug }}" = "true" ]; then
+            echo "🔍 Deployment configuration:"
+            echo "  S3 Prefix: $(cat $GITHUB_OUTPUT | grep s3-prefix | cut -d'=' -f2-)"
+            echo "  Deployment URL: $(cat $GITHUB_OUTPUT | grep deployment-url | cut -d'=' -f2-)"
+          fi
+
+      - name: Configure multi-brand matrix
+        id: brand-config
+        run: |
+          if [ -n "${{ inputs.brand-config }}" ]; then
+            echo "matrix=${{ inputs.brand-config }}" >> $GITHUB_OUTPUT
+            echo "📱 Multi-brand deployment configured"
+          else
+            echo 'matrix={"brand":["default"]}' >> $GITHUB_OUTPUT
+            echo "📱 Single brand deployment"
+          fi
+
+  # Build and test the application
+  build:
+    name: 🏗️ Build Application
+    runs-on: ubuntu-latest
+    if: inputs.skip-build == false
+    needs: [prepare]
+    strategy:
+      matrix: ${{ fromJSON(needs.prepare.outputs.brand-matrix) }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: ${{ inputs.package-manager }}
+
+      - name: Install dependencies
+        run: |
+          debug=""
+          if [ "${{ inputs.debug }}" = "true" ]; then
+            debug="--verbose"
+          fi
+
+          case "${{ inputs.package-manager }}" in
+            "yarn")
+              if [ "${{ inputs.is-yarn-classic }}" = "true" ]; then
+                yarn install --frozen-lockfile $debug
+              else
+                yarn install --immutable $debug
+              fi
+              ;;
+            "npm")
+              npm ci $debug
+              ;;
+            *)
+              echo "❌ Unsupported package manager: ${{ inputs.package-manager }}"
+              exit 1
+              ;;
+          esac
+
+      - name: Run tests
+        if: inputs.skip-tests == false
+        run: |
+          debug=""
+          if [ "${{ inputs.debug }}" = "true" ]; then
+            debug="--verbose"
+          fi
+          
+          # Check if test script exists
+          if ${{ inputs.package-manager }} run | grep -q "test"; then
+            echo "🧪 Running tests..."
+            ${{ inputs.package-manager }} run test $debug
+          else
+            echo "ℹ️ No test script found, skipping tests"
+          fi
+
+      - name: Build application
```

Comment:
If there are multiple brands this also needs to build the brands concurrently using a matrix